internal/failure/failure.go

```diff
@@ -0,0 +1,268 @@
+// Copyright (c) 2018 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package failure
+
+import (
+	"fmt"
+	"sort"
+	"strconv"
+	"strings"
+	"text/scanner"
+)
+
+type (
+	// Field references a Failure field.
+	Field int
+
+	// ID references a Failure identifier.
+	ID int
+)
+
+const (
+	// Filename references the Filename field of a Failure.
+	Filename Field = iota
```

Comment:
I generally do verbose naming on enums on purpose - this introduces other weirdness:

```
var failure failure.Failure // a struct of type failure.Failure
_ = failure.Filename // is this the enum Field value Filename, or the struct field failure.Filename
```

What is the benefit of removing the `FailureField` prefix?